migrate_rules.py

```diff
@@ -64,62 +64,196 @@ def migrate(rancher_url, rancher_api_token, cluster_id, project_id, insecure):
         "authorization": "Bearer %s" % rancher_api_token
     }
 
-    if project_id:
-        url = "%s/v3/projects/%s:%s/projectalertrules" % (rancher_url, cluster_id, project_id)
-        name = "monitoring-v1-rules-%s-%s" % (cluster_id, project_id)
-    else:
-        url = "%s/v3/clusterAlertRules?clusterId=%s" % (rancher_url, cluster_id)
-        name = "monitoring-v1-rules-%s" % cluster_id
-
-    cluster_alert_rules_response = requests.get(url, headers=headers, verify=verify)
-
-    cluster_alert_rules = json.loads(cluster_alert_rules_response.content)
-
-    rules = []
-
-    for cluster_alert_rule in cluster_alert_rules["data"]:
-        if cluster_alert_rule["creatorId"] and cluster_alert_rule["metricRule"]:
-            metric_rule = cluster_alert_rule["metricRule"]
-            prometheus_expression = get_prometheus_expression(
-                metric_rule["expression"],
-                metric_rule["comparison"],
-                metric_rule["thresholdValue"]
-            )
-            rule = OrderedDict(**{
-                "alert": literal(cluster_alert_rule["name"]),
-                "expr": literal(prometheus_expression),
-                "for": cluster_alert_rule["metricRule"]["duration"],
-                "labels": {
-                    "severity": cluster_alert_rule["severity"]
-                },
-                "annotations": OrderedDict(
-                    message=literal("Query %s is %s %s. Current value {{ $value }}" % (
-                        metric_rule["expression"],
-                        metric_rule["comparison"],
-                        metric_rule["thresholdValue"]
-                    ))
+    # Get System Project
+    projects_url = "%s/v3/projects" % (rancher_url)
+    projects_response = requests.get(projects_url, headers=headers, verify=verify)
+    projects = json.loads(projects_response.content)
+    for project in projects["data"]:
+        if project["clusterId"] != cluster_id:
+            continue
+        if "authz.management.cattle.io/system-project" in project["labels"]: 
+            system_project_id = project["id"]
+            break
+
+    # Get Alertmanager Config
+    alerting_config_url = "%s/v3/project/%s/namespacedSecrets/cattle-prometheus:alertmanager-cluster-alerting" % (rancher_url, system_project_id)
+    alerting_config_response = requests.get(alerting_config_url, headers=headers, verify=verify)
+    alerting_config = json.loads(alerting_config_response.content)
+    alertmanager_yaml = yaml.safe_load(
+        base64_decode(alerting_config["data"]["alertmanager.yaml"])
```

Comment:
This fails if the cluster has not alertmanager config (`KeyError: 'data'`). I think we should handle this more gracefully.

Reply:
Response is `{'baseType': 'error', 'code': 'NotFound', 'message': 'secrets "alertmanager-cluster-alerting" not found', 'status': 404, 'type': 'error'}`

Reply:
Good catch. Seems like it would be a quick fix to just check when `alerting_config_response.status_code == 404` before trying to load the `alertmanager_yaml` here.

If we can't find the Alertmanager Secret, I think the behavior of this script should be to only output PrometheusRules and drop the Namespace / Secret resource in favor on Monitoring V2 creating it for the user.

Reply:
Has been updated for this. Also prints out nice log statements to stderr.